restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java

```diff
@@ -589,6 +595,65 @@ private void invite(final Object message) throws IOException, NumberParseExcepti
 
     }
 
+    /**
+     * FIXME: duplicated code make into static function or something more optimized
+     * Replace headers
+     * @param SipServletRequest message
+     * @param Map<String, ArrayList<String> > headers
+     */
+    private void addHeadersToMessage(SipServletRequest message, Map<String, ArrayList<String> > headers) {
```

Comment:
@abdulazizali77 maybe we should apply more strict rules here about what headers should be possible to be added. We should check if header is RURI or a custom headers (starting with X-).

@deruelle @scottbarstow  do you think we should allow more than that?